### PR TITLE
[GEN 2][Mobile] Move Android, iOS, Flutter documentation for multi-step sign-in

### PR DESCRIPTION
**Pull request by salihgueler**
### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [x] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [x] Swift
- [x] Android
- [x] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/[platform]/build-a-backend/auth/multi-step-sign-in/index.mdx [hunk 1]
```diff
@@ -0,0 +1,1820 @@
...
+  return {
+    props: {
+      platform: context.params.platform,
+      meta
+    }
+  };
+}
+
+<InlineFilter filters={['flutter']}>
+After a user has finished signup, they can proceed to sign in. Amplify Auth signin flows can be multi step processes. The required steps are determined by the configuration you provided when you ran `amplify add auth`. Depending on the configuration, you may need to call various APIs to finish authenticating a user's signin attempt. To identify the next step in a signin flow, inspect the `nextStep` parameter in the signin result.
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/auth/multi-step-sign-in/index.mdx:24]: this command does not exist in gen2

File: src/pages/[platform]/build-a-backend/auth/multi-step-sign-in/index.mdx [hunk 2]
```diff
@@ -0,0 +1,1820 @@
...
+
+<Callout warning>
+<b>New enumeration values</b>
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/auth/multi-step-sign-in/index.mdx:27]:
```suggestion

**New enumeration values**
```

File: src/pages/[platform]/build-a-backend/auth/multi-step-sign-in/index.mdx [hunk 3]
```diff
@@ -0,0 +1,1820 @@
...
+    return _handleSignInResult(result);
+  } on AuthException catch (e) {
+    safePrint('Error confirming custom challenge: ${e.message}');
+  }
+}
+```
+
+<Callout warning>
+
+<b>Special Handling on confirmSignIn</b>
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/auth/multi-step-sign-in/index.mdx:262]:
```suggestion
**Special Handling on `confirmSignIn`**
```

File: src/pages/[platform]/build-a-backend/auth/multi-step-sign-in/index.mdx [hunk 4]
```diff
@@ -0,0 +1,1820 @@
...
+  }
+}
+```
+</InlineFilter>
+
+<InlineFilter  filters={['android']}>
+After a user has finished signup, they can proceed to sign in. Amplify Auth signin flows can be multi step processes. The required steps are determined by the configuration you provided when you ran `amplify add auth`. Depending on the configuration, you may need to call various APIs to finish authenticating a user's signin attempt. To identify the next step in a signin flow, inspect the `nextStep` parameter in the signin result.
+
+<Callout warning>
+<b>New enumeration values</b>
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/auth/multi-step-sign-in/index.mdx:409]:
```suggestion

**New enumeration values**
```

File: src/pages/[platform]/build-a-backend/auth/multi-step-sign-in/index.mdx [hunk 5]
```diff
@@ -0,0 +1,1820 @@
...
+    case AuthSignInStep.done:
+      // Could also check that `result.isSignedIn` is `true`
+      safePrint('Sign in is complete');
+  }
+}
+```
+</InlineFilter>
+
+<InlineFilter  filters={['android']}>
+After a user has finished signup, they can proceed to sign in. Amplify Auth signin flows can be multi step processes. The required steps are determined by the configuration you provided when you ran `amplify add auth`. Depending on the configuration, you may need to call various APIs to finish authenticating a user's signin attempt. To identify the next step in a signin flow, inspect the `nextStep` parameter in the signin result.
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/auth/multi-step-sign-in/index.mdx:406]: nit for gen1 command

File: src/pages/[platform]/build-a-backend/auth/multi-step-sign-in/index.mdx [hunk 6]
```diff
@@ -0,0 +1,1820 @@
...
+                error -> Log.e("AuthQuickstart", "Confirm sign in failed: " + error)
+        );
+```
+
+</Block>
+</BlockSwitcher>
+
+<Callout warning>
+
+<b>Special Handling on ConfirmSignIn</b>
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/auth/multi-step-sign-in/index.mdx:965]: can these bolded callout notes be replaced with markdown?

File: src/pages/[platform]/build-a-backend/auth/multi-step-sign-in/index.mdx [hunk 7]
```diff
@@ -0,0 +1,1820 @@
...
+                        Log.i("AuthQuickstart", "Confirm signIn succeeded");
+                    } else {
+                        Log.i("AuthQuickstart", "Confirm sign in not complete. There might be additional steps: " + result.getNextStep());
+                        // Switch on the next step to take appropriate actions.
+                        // If `signInResult.isSignedIn` is true, the next step
+                        // is 'done', and the user is now signed in.
+                    }
+                },
+                error -> Log.e("AuthQuickstart", "Confirm sign in failed: " + error)
+        );
```
[the remark this answers is not in the document]
Comment (salihgueler) [reply src/pages/[platform]/build-a-backend/auth/multi-step-sign-in/index.mdx:839]: copied the code from previous docs but looks about write with the RxJava indentation.

File: src/pages/[platform]/build-a-backend/auth/multi-step-sign-in/index.mdx [hunk 8]
```diff
@@ -0,0 +1,1826 @@
...
+  );
+}
+```
+### Confirm Signup
+If the next step is `resetPassword`, Amplify Auth requires that the user confirm their email or phone number before proceeding.
+Use the `resendSignUpCode` API to send a new sign up code to the registered email or phone number, followed by `confirmSignUp` 
+to complete the sign up.
+
+See the [confirm sign up](/gen1/[platform]/build-a-backend/auth/enable-sign-in/#register-a-user) docs for more information.
+
```
Comment (timngyn) [inline src/pages/[platform]/build-a-backend/auth/multi-step-sign-in/index.mdx:342]: Should this be a gen1 link?

Comment (salihgueler) [reply src/pages/[platform]/build-a-backend/auth/multi-step-sign-in/index.mdx:342]: Great catch!

File: src/pages/[platform]/build-a-backend/auth/multi-step-sign-in/index.mdx (hunk 6 above)
Comment (salihgueler) [reply src/pages/[platform]/build-a-backend/auth/multi-step-sign-in/index.mdx]: All done.

**Review by aspittel — APPROVED** [review]

File: src/pages/[platform]/build-a-backend/auth/multi-step-sign-in/index.mdx [hunk 9]
```diff
@@ -0,0 +1,1826 @@
+import { getCustomStaticPath } from '@/utils/getCustomStaticPath';
+
+export const meta = {
+  title: 'Multi-step sign-in',
+  description:
+    'Use AWS Cognito Auth plugin to complete a multi step authentication flow',
```
Comment (aspittel) [inline src/pages/[platform]/build-a-backend/auth/multi-step-sign-in/index.mdx:6]: Amazon Cognito

File: src/pages/[platform]/build-a-backend/auth/multi-step-sign-in/index.mdx [hunk 10]
```diff
@@ -0,0 +1,1826 @@
...
+    );
+    return _handleSignInResult(result);
+  } on AuthException catch (e) {
+    safePrint('Error confirming TOTP code: ${e.message}');
+  }
+}
+```
+### Confirm signin with custom challenge
+
+If the next step is `confirmSignInWithCustomChallenge`, Amplify Auth is awaiting completion of a custom authentication challenge. The challenge is based on the Lambda trigger you configured as part of a [custom sign in flow](/[platform]/build-a-backend/auth/sign-in-custom-flow/).
```
Comment (aspittel) [inline src/pages/[platform]/build-a-backend/auth/multi-step-sign-in/index.mdx:230]: AWS Lambda

File: src/pages/[platform]/build-a-backend/auth/multi-step-sign-in/index.mdx [hunk 11]
```diff
@@ -0,0 +1,1826 @@
...
+```
+
+</Block>
+</BlockSwitcher>
+
+<Callout warning>
+
+**Special Handling on `confirmSignIn`**
+
+During a confirmSignIn call if `failAuthentication=true` is returned by the lambda the session of the request gets invalidated by cognito, a NotAuthorizedException is returned and a new signIn call is expected via Amplify.Auth.signIn
```
Comment (aspittel) [inline src/pages/[platform]/build-a-backend/auth/multi-step-sign-in/index.mdx:1662]: the Lambda function

**Review by aspittel — CHANGES_REQUESTED** [review]

**Review by timngyn — APPROVED** [review]

**Review by josefaidt — DISMISSED** [review]

**Review by aspittel — APPROVED** [review]

**Pull request merged**